### PR TITLE
[EventHubs] Replace DateTime by DateTimeOffset

### PR DESCRIPTION
**Pull request by kinelski**
All occurrences of `DateTime` type should be replaced by `DateTimeOffset` in Track Two code.

`TrackOneClient` code must remain the same.

Related issue: #6780

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample1_HelloWorld.cs [hunk 1]
```diff
@@ -50,7 +50,7 @@ await using (var client = new EventHubClient(connectionString, eventHubName))
 
                 Console.WriteLine("The Event Hub has the following properties:");
                 Console.WriteLine($"\tThe path to the Event Hub from the namespace is: { properties.Path }");
-                Console.WriteLine($"\tThe Event Hub was created at: { properties.CreatedAtUtc.ToString("yyyy-MM-dd hh:mm:ss tt") }, in UTC.");
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample1_HelloWorld.cs]: Let's please keep the ", in UTC" text.    That's still relevant to those looking at the data;  while the structure is self-descriptive, that doesn't necessarily translate to what a user is seeing.

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample2_ClientWithCustomOptions.cs [hunk 2]
```diff
@@ -68,7 +68,7 @@ await using (var client = new EventHubClient(connectionString, eventHubName, cli
 
                 Console.WriteLine("The Event Hub has the following properties:");
                 Console.WriteLine($"\tThe path to the Event Hub from the namespace is: { properties.Path }");
-                Console.WriteLine($"\tThe Event Hub was created at: { properties.CreatedAtUtc.ToString("yyyy-MM-dd hh:mm:ss tt") }, in UTC.");
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample2_ClientWithCustomOptions.cs]: Let's please keep the ", in UTC" text. That's still relevant to those looking at the data; while the structure is self-descriptive, that doesn't necessarily translate to what a user is seeing.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/SharedAccessSignature.cs [hunk 3]
```diff
@@ -149,14 +149,14 @@ public SharedAccessSignature(string sharedAccessSignature) : this(sharedAccessSi
         }
 
         /// <summary>
-        ///   nitializes a new instance of the <see cref="SharedAccessSignature" /> class.
+        ///   Initializes a new instance of the <see cref="SharedAccessSignature" /> class.
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/SharedAccessSignature.cs]: Thank you!

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/SharedAccessSignature.cs [hunk 4]
```diff
@@ -149,14 +149,14 @@ public SharedAccessSignature(string sharedAccessSignature) : this(sharedAccessSi
...
         /// </summary>
         ///
         /// <param name="eventHubResource">The Event Hubs resource to which the token is intended to serve as authorization.</param>
         /// <param name="sharedAccessKeyName">The name of the shared access key that the signature should be based on.</param>
         /// <param name="sharedAccessKey">The value of the shared access key for the signagure.</param>
         /// <param name="value">The shared access signature to be used for authorization.</param>
-        /// <param name="expirationUtc">The date and time that the shared access signature expires, in UTC.</param>
+        /// <param name="expirationTime">The date and time that the shared access signature expires.</param>
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/SharedAccessSignature.cs]: I'm not sure that `xxxTime` is the pattern that we want to use.   Though, `expiration` or `expirationDateTime` doesn't fit.   Maybe `signatureExpiration` or something else contextually relevant would be more descriptive?

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/SharedAccessSignature.cs [hunk 5]
```diff
@@ -69,10 +69,10 @@ internal class SharedAccessSignature
         public string SharedAccessKey { get; private set; }
 
         /// <summary>
-        ///   The date and time that the shared access signature expires, in UTC.
+        ///   The date and time that the shared access signature expires.
         /// </summary>
         ///
-        public DateTime ExpirationUtc { get; private set; }
+        public DateTimeOffset ExpirationTime { get; private set; }
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/SharedAccessSignature.cs]: Can we go with `SignatureExpiration` or similar here?

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Core/MessagePropertyName.cs [hunk 6]
```diff
@@ -12,8 +12,8 @@
     ///
     internal static class MessagePropertyName
     {
-        /// <summary>The date and time, in UTC, that a message was enqueued.</summary>
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Core/MessagePropertyName.cs]: Let's please leave the explicit UTC in the comments; it certainly doesn't hurt.   Honestly, outside of Microsoft, I haven't seen anyone really make use of `DateTimeOffSet` so the additional context will be helpful, I feel.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs [hunk 7]
```diff
@@ -139,17 +139,17 @@ internal SystemEventProperties()
             /// </summary>
             ///
             /// <param name="sequenceNumber">The logical sequence number of the event within the partition stream of the Event Hub.</param>
-            /// <param name="enqueuedTimeUtc">The date and time, in UTC, that the event was received by the partition.</param>
+            /// <param name="enqueuedTime">The date and time that the event was received by the partition.</param>
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs]: Let's leave the ", in UTC, " call outs in the comments, please.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/EventPosition.cs [hunk 8]
```diff
@@ -112,18 +112,18 @@ public sealed class EventPosition
 
         /// <summary>
         ///   Corresponds to a specific date and time within the partition to begin seeking an event; the event enqueued after the
-        ///   requested <paramref name="enqueuedTimeUtc" /> will become the current position.
+        ///   requested <paramref name="enqueuedTime" /> will become the current position.
         /// </summary>
         ///
-        /// <param name="enqueuedTimeUtc">The date and time, in UTC, from which the next available event should be chosen.</param>
+        /// <param name="enqueuedTime">The date and time from which the next available event should be chosen.</param>
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/EventPosition.cs]: Let's please keep the ", in UTC," call outs in the comments.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Metadata/PartitionProperties.cs [hunk 9]
```diff
@@ -73,23 +73,23 @@ public sealed class PartitionProperties
         /// <param name="beginningSequenceNumber">The first sequence number available for events in the partition.</param>
         /// <param name="lastSequenceNumber">The sequence number observed the last event to be enqueued in the partition.</param>
         /// <param name="lastOffset">The offset of the last event to be enqueued in the partition.</param>
-        /// <param name="lastEnqueueUtc">The date and time, in UTC, that the last event was enqueued in the partition.</param>
+        /// <param name="lastEnqueuedTime">The date and time that the last event was enqueued in the partition.</param>
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Metadata/PartitionProperties.cs]: Let's please keep the ", in UTC," call outs in the comments.

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneComparer.cs [hunk 10]
```diff
@@ -75,9 +75,35 @@ internal static class TrackOneComparer
                     return false;
                 }
 
-                if (!trackOneEvent.SystemProperties.OrderBy(kvp => kvp.Key).SequenceEqual(trackTwoEvent.SystemProperties.OrderBy(kvp => kvp.Key)))
+                foreach (var property in trackOneEvent.SystemProperties)
                 {
-                    return false;
+                    if (property.Key == TrackOne.ClientConstants.EnqueuedTimeUtcName)
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneComparer.cs]: Good catch!

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneComparer.cs [hunk 11]
```diff
@@ -97,7 +123,7 @@ internal static class TrackOneComparer
                 return false;
             }
 
-            // The only meaningful comparison left is to ensure that the property sets are equivilent,
+            // The only meaningful comparison left is to ensure that the property sets are equivalent,
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneComparer.cs]: Yay, free spelling corrections!

**Review by jsquire — CHANGES_REQUESTED** [review]
Substance of these looks great.    I left a few small comments regarding names and text.   Otherwise, solid!

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample1_HelloWorld.cs (hunk 1 above)
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample1_HelloWorld.cs]: Done.

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample2_ClientWithCustomOptions.cs (hunk 2 above)
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample2_ClientWithCustomOptions.cs]: Done.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Core/MessagePropertyName.cs (hunk 6 above)
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/Core/MessagePropertyName.cs]: Done.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs (hunk 7 above)
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs]: Done.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/EventPosition.cs (hunk 8 above)
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/EventPosition.cs]: Done.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Metadata/PartitionProperties.cs (hunk 9 above)
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/Metadata/PartitionProperties.cs]: Done.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/SharedAccessSignature.cs (hunk 4 above)
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/SharedAccessSignature.cs]: Renamed it to `signatureExpiration`.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/SharedAccessSignature.cs (hunk 5 above)
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/SharedAccessSignature.cs]: Renamed it to `SignatureExpiration`.

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Authorization/SharedAccessSignatureCredentialTests.cs [hunk 12]
```diff
@@ -36,7 +36,7 @@ public void ConstructorValidatesTheSignature()
         public void ConstructorValidatesInitializesProperties()
         {
             var value = "TOkEn!";
-            var signature = new SharedAccessSignature(String.Empty, "keyName", "key", value, DateTime.UtcNow.AddHours(4));
+            var signature = new SharedAccessSignature(String.Empty, "keyName", "key", value, DateTimeOffset.Now.AddHours(4));
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/Authorization/SharedAccessSignatureCredentialTests.cs]: These should be `DateTimeOffset.UtcNow.`    Not a huge deal in the tests, but `UtcNow` = the UTC offset (nothing) where `Now` uses the local timezone offset.  

[reference](https://docs.microsoft.com/en-us/dotnet/api/system.datetimeoffset.now?view=netframework-4.8)

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneSharedAccessTokenTests.cs [hunk 13]
```diff
@@ -35,7 +35,7 @@ public void ConstructorValidatesTheSignature()
         [Test]
         public void ConstructorValidatesTheSignatureValue()
         {
-            var signature = new SharedAccessSignature("audience", "keyName", "key", null, DateTime.UtcNow);
+            var signature = new SharedAccessSignature("audience", "keyName", "key", null, DateTimeOffset.Now);
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneSharedAccessTokenTests.cs]: These should be `DateTimeOffset.UtcNow.`    Not a huge deal in the tests, but `UtcNow` = the UTC offset (nothing) where `Now` uses the local timezone offset.  

[reference](https://docs.microsoft.com/en-us/dotnet/api/system.datetimeoffset.now?view=netframework-4.8)

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubClient/EventHubClientLiveTests.cs [hunk 14]
```diff
@@ -144,7 +144,7 @@ await using (var client = new EventHubClient(connectionString, scope.EventHubNam
                     Assert.That(properties, Is.Not.Null, "A set of properties should have been returned.");
                     Assert.That(properties.Path, Is.EqualTo(scope.EventHubName), "The property Event Hub name should match the scope.");
                     Assert.That(properties.PartitionIds.Length, Is.EqualTo(partitionCount), "The properties should have the requested number of partitions.");
-                    Assert.That(properties.CreatedAtUtc, Is.EqualTo(DateTime.UtcNow).Within(TimeSpan.FromSeconds(10)), "The Event Hub should have been created just about now.");
+                    Assert.That(properties.CreatedAt, Is.EqualTo(DateTimeOffset.Now).Within(TimeSpan.FromSeconds(10)), "The Event Hub should have been created just about now.");
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubClient/EventHubClientLiveTests.cs]: These should be `DateTimeOffset.UtcNow.`    Not a huge deal in the tests, but `UtcNow` = the UTC offset (nothing) where `Now` uses the local timezone offset.  

[reference](https://docs.microsoft.com/en-us/dotnet/api/system.datetimeoffset.now?view=netframework-4.8)

**Review by jsquire — APPROVED** [review]
Looks like we just need a quick pass to change `DateTimeOffset.Now` references to `DateTimeOffset.UtcNow` and we're good to go!

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Authorization/SharedAccessSignatureCredentialTests.cs (hunk 12 above)
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/tests/Authorization/SharedAccessSignatureCredentialTests.cs]: Done.

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneSharedAccessTokenTests.cs (hunk 13 above)
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneSharedAccessTokenTests.cs]: Done.

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubClient/EventHubClientLiveTests.cs (hunk 14 above)
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubClient/EventHubClientLiveTests.cs]: Done.

**Pull request merged**